The app crashes if you try to load a second version of your form

#### Software and hardware versions 
Current master branch

#### Problem description
The app crashes if you try to load a second version of your form

#### Steps to reproduce the problem
You can use any form to reporduce the issue
1. Copy your form to odk/forms/ dir
2. Load your form
3. Edit your form you need to change something eg. a question label)
4. Copy your new form to odk/forms/ dir
5. Click on "Fill Blank Form" button

#### Other information 

> FATAL EXCEPTION: AsyncTask #5
                                                                         Process: org.odk.collect.android, PID: 22501
                                                                         java.lang.RuntimeException: An error occured while executing doInBackground()
                                                                             at android.os.AsyncTask$3.done(AsyncTask.java:300)
                                                                             at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
                                                                             at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
                                                                             at java.util.concurrent.FutureTask.run(FutureTask.java:242)
                                                                             at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
                                                                             at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
                                                                             at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
                                                                             at java.lang.Thread.run(Thread.java:818)
                                                                          Caused by: android.database.sqlite.SQLiteConstraintException: NOT NULL constraint failed: forms.md5Hash (code 1299)
                                                                             at android.database.sqlite.SQLiteConnection.nativeExecuteForChangedRowCount(Native Method)
                                                                             at android.database.sqlite.SQLiteConnection.executeForChangedRowCount(SQLiteConnection.java:904)
                                                                             at android.database.sqlite.SQLiteSession.executeForChangedRowCount(SQLiteSession.java:754)
                                                                             at android.database.sqlite.SQLiteStatement.executeUpdateDelete(SQLiteStatement.java:64)
                                                                             at android.database.sqlite.SQLiteDatabase.updateWithOnConflict(SQLiteDatabase.java:1700)
                                                                             at android.database.sqlite.SQLiteDatabase.update(SQLiteDatabase.java:1646)
                                                                             at org.odk.collect.android.provider.FormsProvider.update(FormsProvider.java:477)
                                                                             at android.content.ContentProvider$Transport.update(ContentProvider.java:330)
                                                                             at android.content.ContentResolver.update(ContentResolver.java:1343)
                                                                             at org.odk.collect.android.tasks.DiskSyncTask.doInBackground(DiskSyncTask.java:170)
                                                                             at org.odk.collect.android.tasks.DiskSyncTask.doInBackground(DiskSyncTask.java:46)
                                                                             at android.os.AsyncTask$2.call(AsyncTask.java:288)
                                                                             at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                                             at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231) 
                                                                             at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112) 
                                                                             at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587) 
                                                                             at java.lang.Thread.run(Thread.java:818) 


It has been implemented in this pr https://github.com/opendatakit/collect/pull/1333/files#diff-192a4e10f695374a43bb446cb1f0b10bL664

--- a/collect_app/src/main/java/org/odk/collect/android/provider/FormsProvider.java
+++ b/collect_app/src/main/java/org/odk/collect/android/provider/FormsProvider.java
@@ -448,7 +448,7 @@ public class FormsProvider extends ContentProvider {
                             String oldFile = update.getString(update
                                     .getColumnIndex(FormsColumns.FORM_FILE_PATH));
 
-                            if (formFile == null || formFile.equalsIgnoreCase(oldFile)) {
+                            if (formFile == null || !formFile.equalsIgnoreCase(oldFile)) {
                                 deleteFileOrDir(oldFile);
                             }
 

